omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java

```diff
@@ -184,4 +204,36 @@ private boolean isOfflineConceptEvent(String eventUuid) {
         final Concept offlineConcept = conceptService.getConceptByName("Offline Concepts");
         return offlineConcept != null && offlineConcept.getSetMembers().contains(concept);
     }
+
+    protected String getFilterForLocalIdentifierSource(String identifier) {
+        Session session = null;
+        try {
+            session = sessionFactory.getCurrentSession();
+            session.beginTransaction();
```

Comment:
Do we really need a transaction as we are reading only?

Reply:
Creating transaction boundaries are always necessary even for read-only operations. No communication with the database can occur outside of a database transaction. If we do not end (i.e commit or rollback) a transaction , the database connection used by this transaction will not be released and your database will run out of the available connections eventually.